docs/src/linear_advection.md

```diff
@@ -127,4 +133,30 @@ using BenchmarkTools
 
 ```@example LinearAdvection
 @benchmark solve(prob_sparse, MPRK43I(1.0, 0.5); save_everystep = false)
-```
\ No newline at end of file
+```
+
+By default, we use an LU factorization for the linear systems. At the time of
+writing, Julia uses [SparseArrays.jl](https://github.com/JuliaSparse/SparseArrays.jl)
+defaulting to UMFPACK from SuiteSparse in this case. However, the linear systems
+do not necessarily have the structure for which UMFPACK is optimized for. Thus,
+it is often possible to gain performance by switching to KLU instead.
+
```

Comment:
Are you saying that this is the reason for the large number of allocations we see in the linear advection tutorial?

There's also something else we should look at. When we initialize the linear solvers we use something like
```julia
linsolve = init(linprob, alg.linsolve, alias_A = true, alias_b = true,
                        assumptions = LinearSolve.OperatorAssumptions(true))
```
The statement `alias_A=true` allows the linear solver to reuse the memory of the system matrix and modify it. The idea behind this was that in an MPRK scheme we no longer need the system matrix once the system is solved, so the linear solver can change it at will.

For a sparse matrix this modification of the system matrix could become problematic, since the sparsity pattern of L and U does not match that of A, especially in the linear advection tutorial, where A is a banded matrix with an entry in the upper right corner, for which in genaral L or U could become a full triangular matrix.

All I want to say is that we should try `alias_A=false ` and check if we see any difference.

Reply:
Shall we use `alias_A = !issparse(the_matrix_we_put_in)` for the general case?